chore: Update plugin test config and project tests

Gustl22 opened the pull request:
~Some configurations are inherted from other configs. So in that case the "PRODUCT_BUNDLE_IDENTIFIER" is not present. Therefore exits [resolved_build_setting](https://github.com/CocoaPods/Xcodeproj/blob/ab3dfa504b5a97cae3a653a8924f4616dcaa062e/lib/xcodeproj/project/object/native_target.rb#L54).~

Edit:
- Update outdated plugin test config options
- Update test project to not fail
- Update README to support command line
- Minor optimization on assigning `target`
- Update generated files

Gustl22 committed:
fix: Resolve inherited bundle id in build settings

lyndsey-ferguson commented:
Thank you for the contribution offer! Can you add a test to the spec here?
https://github.com/lyndsey-ferguson/get_product_bundle_id/blob/master/spec/get_product_bundle_id_action_spec.rb

Gustl22 committed:
tests + revert

Gustl22 committed:
Adapt readme

Gustl22 commented:
Thank you for pointing this out. Somehow the application didn't have set the bundle identifier on all build configurations. I orginally thought, this failure comes from not able to be resolving inherited configs from `baseConfigurationReference` but that's not the case.

Now I updated some of the configs which were outdated and updated the tests of the project itself (not the plugin) to not fail (if that's desired).
